lineapy/api/models/linea_artifact.py

```diff
@@ -105,36 +103,118 @@ def get_value(self) -> object:
         """
         Get and return the value of the artifact
         """
-        pickle_filename = self.db.get_node_value_path(
-            self._node_id, self._execution_id
-        )
-        if pickle_filename is None:
+        metadata = self.get_metadata()
+        linea_metadata = metadata["lineapy"]
+        saved_filepath = linea_metadata.storage_path
+        if saved_filepath is None:
             return None
         else:
-            # TODO - set unicode etc here
             track(GetValueEvent(has_value=True))
+            # read from mlflow
+            if "mlflow" in metadata.keys():
+                return read_mlflow(metadata["mlflow"])
+
+            # read from lineapy
+            return self._read_pickle(saved_filepath)
+
+    @lru_cache(maxsize=None)
+    def _get_storage_path(self) -> Optional[str]:
+        return self.db.get_node_value_path(self._node_id, self._execution_id)
+
+    @lru_cache(maxsize=None)
+    def get_metadata(self, lineapy_only: bool = False) -> ArtifactInfo:
+        """
+        Get artifact backend storage metadata
+
+        :param lineapy_only: If ``False``, will include both LineaPy related
+            metadata and metadata from storage backend(if it is not LineaPy).
+            If ``True``, will only return LineaPy related metadata no matter
+            which storage backend is using.
+
+        :return: Metadata for artifact backend storage.
+
+        """
 
-            artifact_storage_dir = options.safe_get("artifact_storage_dir")
-            filepath = (
-                artifact_storage_dir.joinpath(pickle_filename)
-                if isinstance(artifact_storage_dir, Path)
-                else f'{artifact_storage_dir.rstrip("/")}/{pickle_filename}'
+        if self._artifact_id is None or self.date_created is None:
+            artifactorm = self.db.get_artifactorm_by_name(
+                artifact_name=self.name, version=self.version
             )
-            try:
-                logger.debug(
-                    f"Retriving pickle file from {filepath} ",
-                )
-                return read_pickle(
-                    filepath, storage_options=options.get("storage_options")
+            self._artifact_id = artifactorm.id
+            self.date_created = artifactorm.date_created
+
+        assert isinstance(self._artifact_id, int)
+        assert isinstance(self.date_created, datetime)
+
+        storage_path = self._get_storage_path()
+        storage_backend = (
```

Comment:
nitpick, but can we break this down into a proper if/else statement? 

I would expect this logic will be more complicated with more integrations so we might as well write this out ...